app/main.py

```diff
@@ -1,2 +1,59 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    class Node:
+        def __init__(self, key: Hashable, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.hash = hash(key)
+
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size >= self.capacity * self.load_factor:
+            self._resize()
+
+        index = self._hash(key)
```

Comment:
The _hash method is used here to determine the index. Ensure that this method is consistent with the task requirements for hashing. If the task specifies a different hashing mechanism, it should be implemented in the _hash method.